tests/Teams/ActionBasedMessagingExtension-FetchTask/Bots/ActionBasedMessagingExtensionFetchTaskBot.cs

```diff
@@ -17,5 +20,110 @@ protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivi
         {
             await turnContext.SendActivityAsync(MessageFactory.Text($"echo: {turnContext.Activity.Text}"), cancellationToken);
         }
+
+        protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionFetchTaskAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+        {
+            var reply = MessageFactory.Text("OnTeamsMessagingExtensionFetchTaskAsync MessagingExtensionQuery: " + JsonConvert.SerializeObject(query));
+            await turnContext.SendActivityAsync(reply, cancellationToken);
+
+            return new MessagingExtensionActionResponse
+            {
+                Task = new TaskModuleContinueResponse
+                {
+                    Value = new TaskModuleTaskInfo()
+                    {
+                        Card = new Attachment
+                        {
+                            Content = GetTaskModuleAdaptiveCard(),
+                            ContentType = AdaptiveCard.ContentType,
+                        },
+                        Height = 450,
+                        Width = 450,
+                        Title = "Task Module Example",
+                    },
+                },
+            };
+        }
+
+        protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionSubmitActionAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action, CancellationToken cancellationToken)
+        {
+            var reply = MessageFactory.Text("OnTeamsMessagingExtensionSubmitActionAsync MessagingExtensionAction: " + JsonConvert.SerializeObject(action));
+            await turnContext.SendActivityAsync(reply, cancellationToken);
+
+            var adaptiveCard = GetCardFromSubmitExampleData(JsonConvert.DeserializeObject<SubmitExampleData>(action.Data.ToString()));
+
+            return new MessagingExtensionActionResponse
```

Comment:
Is this behavior in a screenshot in the README? I don't think I saw it there...

Reply:
Step 4?